File: leetcode/0474.cpp
class Solution {
public:
    int findMaxForm(vector<string>& strs, int m, int n) {
        vector<vector<int>> dp(m + 1, vector<int>(n + 1, 0));
        static auto _ = []() {
            ios::sync_with_stdio(false);
            cin.tie(nullptr);
            cout.tie(nullptr);
            return 0;
        }();
        auto countoneanszero = [](auto&& s) {
            int a = 0, b = 0;
            for (auto&& c : s)
                if (c == '0')
                    a++;
                else
                    b++;
            return make_tuple(a, b);
        };
        for (auto&& str : strs) {
            auto&& [a, b] = countoneanszero(str);
            for (int i = m; i >= a; --i) {
                for (int j = n; j >= b; --j) {
                    dp[i][j] = max(dp[i][j], dp[i - a][j - b] + 1);
                }
            }
        }
        return dp[m][n];
    }
};

File: leetcode/0474.go
package leetcode

func findMaxForm(strs []string, m int, n int) int {
	dp := make([][]int, m+1)
	for i := range dp {
		dp[i] = make([]int, n+1)
	}

	for i := range strs {
		a, b := 0, 0
		for c := range strs[i] {
			if strs[i][c] == '0' {
				a++
			} else {
				b++
			}
		}

		for j := m; j >= a; j-- {
			for k := n; k >= b; k-- {
				dp[j][k] = max(dp[j][k], dp[j-a][k-b]+1)
			}
		}
	}
	return dp[m][n]
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}
